shared/js/dialer/keypad.js

```diff
@@ -836,5 +852,56 @@ var KeypadManager = {
         self._vibrationEnabled = !!value;
       });
     });
+  },
+
+  _updateCaretPosition: function kh_updateCaretPosition() {
+    this._insertPosition = this._realStartPosition();
+  },
+
+  _deleteDigitAtCaret: function kh_deleteDigitAtCaret() {
+    if (this._insertPosition === null) {
+      this._phoneNumber = this._phoneNumber.slice(0, -1);
+    } else {
+      var start = this._realStartPosition();
+      var end = this._realEndPosition();
+      if (start > 0 && start === end) {
+        start = end - 1;
+      }
+      this._phoneNumber = this._phoneNumber.substring(0, start) +
+                          this._phoneNumber.substring(end);
+      this._insertPosition = start;
```

Comment:
When we clear the number the caret remains visible until we type a new number. Can you change this so that once ``this._phoneNumber`` is empty the caret is hidden?